src/main/java/edu/harvard/iq/dataverse/DatasetFieldServiceBean.java

```diff
@@ -601,10 +601,16 @@ private JsonObject filterResponse(JsonObject cvocEntry, JsonObject readObject, S
                         job.add(filterKey, termUri);
                     } else if (pattern.contains("{")) {
                         if (pattern.equals("{0}")) {
-                            if (vals.get(0) instanceof JsonArray) {
-                                job.add(filterKey, (JsonArray) vals.get(0));
-                            } else {
-                                job.add(filterKey, (String) vals.get(0));
+                            if (!vals.isEmpty() && vals.get(0) != null) {
+                                if (vals.get(0) instanceof JsonArray) {
+                                    job.add(filterKey, (JsonArray) vals.get(0));
+                                }
+                                else if (vals.get(0) instanceof String) {
+                                    job.add(filterKey, (String) vals.get(0));
+                                }
+                                else {
```

Comment:
This is either an error or it is not. If it is, then it should result in an exception. Otherwise, it should not be logged as a warning.